src/backend/core/models.py

```diff
@@ -325,3 +363,73 @@ def clean_fields(self, exclude=None):
         else:
             raise ValidationError({"name": f'Room name "{self.name:s}" is reserved.'})
         super().clean_fields(exclude=exclude)
+
+
+# todo - discuss how the path could changed, and we could loose track of file
+# todo - discuss the uniqueness of worker_id field
+class Recording(BaseModel):
+    """Model for recordings that take place in a room"""
+
+    creator = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        related_name="recordings",
+        verbose_name=_("Creator"),
+    )
+    room = models.ForeignKey(
+        Room,
+        on_delete=models.CASCADE,
+        related_name="recordings",
+        verbose_name=_("Room"),
+    )
+    status = models.CharField(
+        max_length=20,
+        choices=RecordingStatusChoices.choices,
+        default=RecordingStatusChoices.ACTIVE,
+    )
+    mode = models.CharField(
+        max_length=20,
+        choices=RecordingModeChoices.choices,
+        default=RecordingModeChoices.SCREEN_RECORDING,
+        verbose_name=_("Recording mode"),
+        help_text=_("Defines the type of recording being performed."),
+    )
```

Comment:
Note from our IRL discussion:

I would replace the mode, with a worker_kind, like `screen-recording-livekit-egress-worker` where the kind is an HRID declared by the worker:

``` python
class LivekitEgressWorker(AbstractRecordingWorker):
    # ...


class ScreenRecordingLivekitEgressWorker(LivekitEgressWorker):
    hrid = screen-recording-livekit-egress-worker
    # ...
```

So from a recording you can get the worker class directly

Reply:
Mode field is retained, but worker selection is handled by the backend. This keeps client logic simple - clients specify the mode they want, and backend matches appropriate workers.